### DISCUSSION TITLE
question: are there optimize options in power fx?

### DISCUSSION THREAD

**Discussion by hiber-wang** [microsoft/Power-Fx#811] [Q&A] [answered]
Hi, 
Can I use different optimize mode to compile power fx program?

**Comment by MikeStall** [comment, marked as answer]
Was there a particular option  you were looking for? 

The Power Fx nugets themselves are compiled retail and optimized by .net. 
The execution engine that evaluates the user's expression has some optimizations. (ie, some constant folding, etc) 

But there are not options that the expression author passes in w/ their expression .  Fx will always run it the best it can.

**Reply by hiber-wang** [reply]
Thanks for your answer！
I am looking for the execution engine optimize options, and your answer was very helpful !
I would also like to ask whether I have some way to close these optimizations?

**Reply by MikeStall** [reply]
What do you mean by "close these optimizations"?

**Reply by hiber-wang** [reply]
@MikeStall 
Thank you for your reply! I mean can I execute the expression without any optimizations by changing the source code or some other ways?

**Reply by MikeStall** [reply]
There is no flag you can set to change Power Fx codegen. 

Why do you want to disable all optimizations?
